File: week4_tutorial/IPC/message passing/go/tcp_client.go
package main

import (
	"fmt"
	"net"
)

func main() {
	// Replace with the actual IP address of the server
	serverIP := "server_ip_here"
	serverPort := 12345

	// Establish a TCP connection to the server
	conn, err := net.Dial("tcp", fmt.Sprintf("%s:%d", serverIP, serverPort))
	if err != nil {
		fmt.Println("Error connecting:", err)
		return
	}
	defer conn.Close() // Close the connection when the function exits

	// Message to send to the server
	message := "Hello from Go Client!"

	// Send the message as a byte array over the connection
	_, err = conn.Write([]byte(message))
	if err != nil {
		fmt.Println("Error sending message:", err)
		return
	}
}

File: week4_tutorial/IPC/message passing/rust/tcp_client.rs
use std::io::Write;
use std::net::TcpStream;

fn main() {
    // Replace with the actual IP address of the server
    let server_ip = "server_ip_here";
    let server_port = 12345;

    // Attempt to establish a TCP connection to the server
    match TcpStream::connect(format!("{}:{}", server_ip, server_port)) {
        Ok(mut stream) => {
            // Message to send to the server
            let message = "Hello from Rust Client!";
            
            // Write the message as bytes to the stream
            match stream.write(message.as_bytes()) {
                Ok(_) => {
                    // Message sent successfully
                    println!("Message sent: {}", message);
                }
                Err(e) => {
                    // Error occurred while sending the message
                    println!("Error sending message: {:?}", e);
                }
            }
        }
        Err(e) => {
            // Error occurred while establishing the connection
            println!("Error connecting: {:?}", e);
        }
    }
}
